backend/src/store.py

```diff
@@ -57,6 +57,17 @@ def get_random(cls) -> Path:
 
         return Path(random.choice(files))
 
+    @classmethod
+    def get_by_id(cls, result_id: str) -> Path | None:
```

Comment:
for `or None` typing has [Optional](https://docs.python.org/3/library/typing.html#typing.Optional)

Reply:
I think using union types are the "modern" way of doing this
https://discuss.python.org/t/clarification-for-pep-604-is-foo-int-none-to-replace-all-use-of-foo-optional-int/26945/6